### PR TITLE
format: don't format files in third_party

### PR DESCRIPTION
**Pull request by piscisaureus**
It's annoying, and it also makes appveyor slow because it will re-upload
the third_party cache every time something changes in there.

### CODE REVIEW

File: tools/util.py [hunk 1]
```diff
@@ -130,16 +130,27 @@ def touch(fname):
...
+def str_to_list(value):
+    return value if type(value) is list else [value]
+
+
 # Recursive search for files of certain extensions.
 # (Recursive glob doesn't exist in python 2.7.)
-def find_exts(directory, *extensions):
+def find_exts(directories, extensions, skip=[]):
+    directories = str_to_list(directories)
+    extensions = str_to_list(extensions)
```
Comment (ry) [inline tools/util.py]: Just have directories and extensions always be a list.
```
isinstance(directories, list)
isinstance(extensions, list)
```

File: tools/util.py [hunk 2]
```diff
@@ -130,16 +130,27 @@ def touch(fname):
         open(fname, 'a').close()
 
 
+def str_to_list(value):
+    return value if type(value) is list else [value]
```
Comment (ry) [inline tools/util.py]: Remove - no need for ergonomics that add complexity in this internal tooling.

File: tools/format.py [hunk 3]
```diff
@@ -17,31 +17,19 @@
...
-for fn in ["BUILD.gn", ".gn"] + find_exts("build_extra", ".gn", ".gni"):
+for fn in ["BUILD.gn", ".gn"] + find_exts("build_extra", [".gn", ".gni"]):
     run(["third_party/depot_tools/gn", "format", fn], env=google_env())
 
-# We use `glob()` instead of `find_exts()` in the tools directory, because:
-#   * On Windows, `os.walk()` (called by `find_exts()`) follows symlinks.
-#   * The tools directory contains a symlink 'clang', pointing at the directory
-#     'third_party/v8/tools/clang', which contains many .py files.
-#   * These third party python files shouldn't be formatted.
-#   * The tools directory has no subdirectories, so `glob()` is sufficient.
```
Comment (ry) [inline tools/format.py]: Move this comment to `find_exts` definition.

Comment (piscisaureus) [reply tools/format.py]: ✔️

File: tools/util.py (hunk 2 above)
Comment (piscisaureus) [reply tools/util.py]: ok done

File: tools/util.py (hunk 1 above)
Comment (piscisaureus) [reply tools/util.py]: ✌️

**Pull request merged**